text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,545 @@
+
+1. **Schedule**: A schedule is the main resource you create, configure, and manage using Amazon EventBridge Scheduler. Every schedule has a schedule expression that determines when, and with what frequency, the schedule runs. EventBridge Scheduler supports three types of schedules: rate, cron, and one-time schedules. When you create a schedule, you configure a target for the schedule to invoke. 
+2. **Targets**: A target is an API operation that EventBridge Scheduler calls on your behalf every time your schedule runs.  Targets can include Amazon EC2 instances, AWS Lambda functions, Kinesis streams, Amazon ECS tasks, Step Functions state machines, Amazon SNS topics, Amazon SQS queues, Amazon CloudWatch LogGroups, and built-in targets. A target receives events in JSON format.
+3. **Schedule Group**: Scheduling groups help you organize your schedules. Scheduling groups support tags that you can use for cost allocation, access control, and resource organization. When creating a new schedule, you can add it to a scheduling group.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project. It allows you to define Event Bridge Schedules.
+
+
+## Defining a schedule 
+
+```ts
+
+const target = new targets.LambdaInvoke(props.func, {
+    input: ScheduleTargetInput.fromObject({
+        "payload": "useful"
+    })
+});
+    
+const schedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test schedule that invokes lambda function every 10 minutes.',
+});
+```
+
+### Schedule Expressions
+
+Both rate-based and cron-based schedules are recurring schedules. You configure each recurring schedule type using a schedule expression for the type of schedule you want to configure, and specifying a time zone in which EventBridge Scheduler evaluates the expression.
+
+```ts
+const rateBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test rate-based schedule',
+});
+
+const cronBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.cron({ day: '20', month: '11' }),
+    target,
+    scheduleTimeZone: TimeZone.AMERICA_NEW_YORK,
+    description: 'This is a test cron-based schedule in New York timezone',
+});
+```
+
+A one-time schedule is a schedule that invokes a target only once. You configure a one-time schedule when by specifying the time, date, and time zone in which EventBridge Scheduler evaluates the schedule.
+
+```ts
+const oneTimeSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.at(new Date(2022, 10, 20, 19, 20, 23)),
+    target,
+    scheduleTimeZone: TimeZone.AMERICA_NEW_YORK,
```

Comment:
I need to check. Because I am reusing a class that already has `cron`, I only added `at`.

Reply:
>  seems a bit odd to have schedule and scheduleTimeZone since scheduleTimeZone seems to be operating on whatever information is in schedule

Yeah, I agree it's odd. However, I don't think we have any other option if we want to reuse class `Schedule` from `events` module, as class `Schedule` already has a method `cron` that does not accept a timezone. Alternatively, we could create our own class `ScheduleExpression` that does not inherit from `Schedule`. What do you think?

Reply:
Lets just go with as is for now. I'm fine with `scheduleTimeZone`.